syntax = "proto2";
option optimize_for = SPEED;

// 服务器类型
enum enServerType
{
	FE_GAMESERVER			= 0;		// 代理服务器
	FE_PROXYSERVER			= 1;		// 代理服务器
	FE_WEBSERVER			= 2;		// http服务器
};

enum enMessageCmd
{	
	MESS_LOGIC          = 0;    // 逻辑cmd
	MESS_REGISTER		= 1;	// 注册
	MESS_KEEPALIVE		= 2;	// 心跳
}

/////////////////////////////////(dbserver,gameserver等等) <==> ProxyServer/////////////////////
//发送给proxyserver的消息头
message CProxyHead
{
	required	uint32			srcFE			= 1;							// 源服务器类型
	required	uint32			srcID			= 2;							// 源服务器ID
	required	uint32			dstFE			= 3;							// 目标服务器类型
	required	uint32			dstID			= 4;							// 目标服务器ID
	required	uint64			timeStamp		= 5;							// 时间戳(毫秒)
	optional	enMessageCmd	opFlag			= 6;							// 指令类型
};

// 单个消息
message CProxyMessage
{
	enum Msg
	{
	    MsgID	= 10000;
	}
	required	CProxyHead		msgHead	= 1;						// 消息头+
	optional	fixed64			msgPara	= 2[default = 0];	// 消息体( 指针 4bytes ) 消息传递时是无效的，这是服务器内部自己赋值的，方便获取消息
};